packages/ssh-tunnel/src/index.ts

```diff
@@ -195,7 +147,55 @@ export class SshTunnel extends EventEmitter {
     }
   }
 
+  private async connectSsh(stayConnected = false): Promise<void> {
+    this.stayConnected = stayConnected || this.stayConnected;
+
+    if (this.connected) {
+      debug('already connected');
+      return;
+    }
+
+    if (this.connectingPromise) {
+      debug('reusing connectingPromise');
+      return this.connectingPromise;
+    }
+
+    if (!stayConnected) {
+      // A socks5 request could come in after we deliberately closed the connection. Don't reconnect in that case.
+      throw new Error('Disconnected.');
+    }
+
+    debug('creating SSH connection');
+    const ac = new AbortController();
```

Comment:
The AbortController allows us to cancel listening to the 'error' and 'ready' events when the Promise.race resolves or rejects. A but minor in the bigger scheme of things, but would technically be a leak?

Reply:
nice! didn't know `once` accepted a signal

Reply:
It’s only supported in Node.js 15+, I think this might not be working yet? But I guess it’s fine to leave it like this and see it start working when we upgrade Electron :)

Reply:
I think it is fine this way - the leaks were already there and they are super minor. You're only going to make so many ssh tunnels.

Reply:
Actually I just removed the AbortController stuff for now because it was spiralling out and out. It requires jsdom so I'd have to systematically move more and more packages onto our react mocha config so that the polyfill is there when it uses ssh-tunnel.